### PR TITLE
docs: Add contributing documentation

### PR DESCRIPTION
**Pull request by dmurray-lacework**
issue: https://lacework.atlassian.net/browse/ALLY-661

Add contributing documentation
- CONTRIBUTING.md
- DEVELOPER_GUIDELINES.md
- ISSUE_TEMPLATE/bug_report.md
- ISSUE_TEMPLATE/feature_request.md 
- ISSUE_TEMPLATE/pull_request_template.md 

Add `make help`
```
❯ make help    
-------------------------------------------------------------------
Lacework terraform-aws-cloudtrail Makefile helper:

 ci - *CI ONLY* Runs tests on CI pipeline
 release - *CI ONLY* Prepares a release of the Terraform module
-------------------------------------------------------------------
```

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Please complete the following information):**
+ - Terraform Version: [e.g. v1.0.0 ]
+ - Module Version [e.g. v0.15.0]
```
[the remark this answers is not in the document]
Comment (afiune) [reply .github/ISSUE_TEMPLATE/bug_report.md:22]: Also, could we add a command to help users get this information? Maybe `terraform version` or `terraform providers`??? (I actually don't know)

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+name: Feature request
+about: Suggest an idea for this project
+title: 'feat: '
+labels: 'feat'
+---
+
+# Feature Request
+
+**Describe the Feature Request**
+<!-- A clear and concise description of what the feature request is. Please include if your feature request is related to a problem -->
```
Comment (afiune) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]: Why these comments are surrounded by `<!-- -->`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Include as much relevant information as possible. See the [bug template](https://github.com/lacework/terraform-aws-cloudtrail/blob/main/.github/ISSUE_TEMPLATE/bug_report.md) for help on creating a new issue.
+
+## Feature Requests
+
+If you wish to submit a request to add new functionality or an improvement to a terraform module then use the the [feature request](https://github.com/lacework/terraform-aws-cloudtrail/blob/main/.github/ISSUE_TEMPLATE/feature_request.md) template to 
+open a new [issue](https://github.com/lacework/go-sdk/issues/new)
+
+## Creating a Pull Request
+
+If you have made a change or added new functionality, you can submit pull request. The project maintainers will aim to review in a reasonable timeframe. When submitting a pull request please read the [developer guidelines](/DEVELOPER_GUIDELINES.md)
```
Comment (afiune) [inline CONTRIBUTING.md:33]: "e need to define what "reasonable timeframe" means, a week? would that be reasonable? 2 weeks?

**Review by afiune — COMMENTED** [review]
This is looooooking so good!

File: CONTRIBUTING.md (hunk 3 above)
Comment (dmurray-lacework) [reply CONTRIBUTING.md:33]: I think 2 weeks is reasonable

Comment (afiune) [reply CONTRIBUTING.md:33]: I agree, and we can always adjust it but at least, it is a bit more clear.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply .github/ISSUE_TEMPLATE/bug_report.md:22]: Sounds good will do

File: GNUmakefile [hunk 4]
```diff
@@ -1,10 +1,23 @@
...
+.PHONEY: ci
+ci: ## *CI ONLY* Runs tests on CI pipeline
 	scripts/ci_tests.sh
 
-release: ci
+.PHONEY: release
+release: ci ## *CI ONLY* Prepares a release of the Terraform module
 	scripts/release.sh prepare
+
+
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply GNUmakefile:23]: A terraform linter? As far as I know terraform doesn't have a native linter. We'd have to say to install tflint or equiv tool.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+name: Feature request
+about: Suggest an idea for this project
+title: 'feat: '
+labels: 'feat'
+---
+
+# Feature Request
+
+**Describe the Feature Request**
+ A clear and concise description of what the feature request is. Please include if your feature request is related to a problem
```
Comment (afiune) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]:
```suggestion
A clear and concise description of what the feature request is. Please include if your feature request is related to a problem
```

File: DEVELOPER_GUIDELINES.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+main.tf - Store the main structure of your terraform code in this file
+variables.tf - All variables for your project
+
+output.tf - All outputs in this file
+
+tfvars.example - An example terraform.tfvars file for easy cp for users (note: *.tfvars are typically 
+ignored by .gitignore
+
+.gitignore - Critical to ensure that any sensitive information used in tfvars are not checked in to git
```
Comment (afiune) [inline DEVELOPER_GUIDELINES.md:80]:
```suggestion
* `main.tf` - Store the main structure of your terraform code in this file
* `variables.tf` - All variables for your project
* `output.tf` - All outputs in this file
* `tfvars.example` - An example terraform.tfvars file for easy cp for users (note: *.tfvars are typically 
ignored by .gitignore
* `.gitignore` - Critical to ensure that any sensitive information used in tfvars are not checked in to git
```

File: DEVELOPER_GUIDELINES.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  use_existing_iam_role = true
+  iam_role_name         = module.aws_config.iam_role_name
+  iam_role_arn          = module.aws_config.iam_role_arn
+  iam_role_external_id  = module.aws_config.external_id
+}
+```
+
+The example above will work for version 0.1.9 as well as 0.4.0, but will not pull in any major releases such as 1.0.0.
+
+For more information visit Semantic Versioning 2.0.0
```
Comment (afiune) [inline DEVELOPER_GUIDELINES.md:111]:
```suggestion
For more information visit [Semantic Versioning 2.0.0](https://semver.org/) 
```

**Review by afiune — APPROVED** [review]
two nits that look better (IMO) - Thanks for documenting all of this, lets merge and see how the template looks, if they look good, let's deploy on all modules!! 💯

tenor-181752422

**Review by declanwilson-lw — CHANGES_REQUESTED** [review]
I think the pull request template can come out of the ISSUES_TEMPLATES sub dir, and into .github/ ?
examples of PR templates:
https://github.com/lacework/services/blob/main/.github/pull_request_template.md

loads of templates out there for PRs... not sure what best works for tech-ally ?

File: DEVELOPER_GUIDELINES.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+* `variables.tf` - All variables for your project
+* `output.tf` - All outputs in this file
+* `tfvars.example` - An example terraform.tfvars file for easy cp for users (note: *.tfvars are typically 
+ignored by .gitignore
+* `.gitignore` - Critical to ensure that any sensitive information used in tfvars are not checked in to git
+
+
+***Version Support / Documentation***
+
+Hashicorp release frequent patch and minor updates as needed, as well as new major releases of Terraform each year. Although Hashicorp provide solid documentation on how to upgrade between major releases of Terraform, Lacework must contend with the fact that Lacework customers do not all upgrade in a timely manor. For this reason Tech Ally must continue to update documentation with supported versions of Terraform, as well as update CI pipelines to test changes across each supported version.
```
Comment (declanwilson-lw) [inline DEVELOPER_GUIDELINES.md:81]: Tech Ally = Tech Alliance team ?

Comment (dmurray-lacework) [reply DEVELOPER_GUIDELINES.md:81]: I can update this to use the full name.

**Review by declanwilson-lw — APPROVED** [review]
lgtm - nice!

**Pull request merged**